apps/emqx_gateway/src/emqx_gateway_api_listeners.erl

```diff
@@ -80,7 +83,39 @@ paths() ->
 
 listeners(get, #{bindings := #{name := Name0}}) ->
     with_gateway(Name0, fun(GwName, _) ->
-        {200, emqx_gateway_conf:listeners(GwName)}
+        Listeners = emqx_gateway_conf:listeners(GwName),
+        ListenOns = lists:map(
+            fun(#{id := Id, <<"bind">> := BinListenOn}) ->
+                {Id, erlang:binary_to_integer(BinListenOn)}
```

Comment:
Is there have a crash error due to the `BinListenOn` is a `integer()` or `ip_port()` type?

Reply:
nit: Maybe we can move the L86-L118 into emqx_gateway_http module to get more short lines for here🤔?